src/bindings/fluid_cmd.c

```diff
@@ -1086,275 +1086,367 @@ fluid_handle_reverbpreset(void *data, int ac, char **av, fluid_ostream_t out)
+        {
+            fluid_ostream_printf(out, "%s: unit index \"%s\" must be in range [%d..%d]\n",
+                                 name_cde, av[0], 0, nfxunits - 1);
+            return -2;
+        }
     }
+    return fxunit_idx;
+}
+
+/* parameter value */
+struct value
+{
+    char *name;
+    fluid_real_t min;
+    fluid_real_t max;
+};
 
-    if(room_size > 1.0)
+/*
+  check 2 arguments for reverb commands : unit fx index  , value
+  - unit index must be an integer in the range [-1..synth->effects_groups].
+  - value must be a double in the range [min..max]
+  @param param a pointer on a value to return the second value argument.
+  return the unit index:
+  -1 when the command is for all fx unit.
+  0 to synth->effects_groups-1 when the command is for this unit index.
+  -2 if error.
+*/
+static int check_fx_reverb_param(int ac, char **av, fluid_ostream_t out,
+                       fluid_synth_t *synth, const char *name_cde,
+                       const struct value *value,
+                       fluid_real_t *param)
+{
+    /* get and check unit index argument */
+    int fxunit_idx = check_fx_unit_idx(ac, av, out, synth, name_cde);
+    if(fxunit_idx >= -1)
     {
-        fluid_ostream_printf(out, "rev_setroomsize: Room size too big!\n");
-        return FLUID_FAILED;
-    }
+        fluid_real_t val;
 
-    fluid_synth_set_reverb_roomsize(handler->synth, room_size);
-    return FLUID_OK;
+        /* get and check value argument */
+        ac--;
+        val = atof(av[ac]);
+        if( !fluid_is_number(av[ac]) || val < value->min || val > value->max)
+        {
+            fluid_ostream_printf(out, "%s: %s \"%s\" must be in range [%f..%f]\n",
+                        name_cde, value->name, av[ac], value->min, value->max);
+            return -2;
+        }
+        *param = val;
+    }
+    return fxunit_idx;
 }
 
+/* reverb commands enum */
+enum reverb_cde
+{
+    REVERB_ROOMSIZE_CDE,
+    REVERB_DAMP_CDE,
+    REVERB_WIDTH_CDE,
+    REVERB_LEVEL_CDE,
+    NBR_REVERB_CDE
+};
+
 /* Purpose:
- * Response to 'rev_setdamp' command.
- * Load the new damp factor into the reverb unit. */
-int
-fluid_handle_reverbsetdamp(void *data, int ac, char **av, fluid_ostream_t out)
+ * Response to fluid_handle_reverbsetxxxx commands
+ */
+static int
+fluid_handle_reverb_command(void *data, int ac, char **av, fluid_ostream_t out,
+                            enum reverb_cde cde )
 {
-    FLUID_ENTRY_COMMAND(data);
-    fluid_real_t damp;
+    /* reverb commands name table */
+    static const char *name_cde[NBR_REVERB_CDE] =
+    {"rev_setroomsize", "rev_setdamp", "rev_setwidth", "rev_setlevel"};
 
-    if(ac < 1)
+    /* name and min/max values table */
+    static struct value values[NBR_REVERB_CDE] =
     {
-        fluid_ostream_printf(out, "rev_setdamp: too few arguments.\n");
-        return FLUID_FAILED;
-    }
+        {"room size", 0.0F, 1.0F}, {"damp", 0.0F, 1.0F},
+        {"width", 0.0F, 100.0F},   {"level", 0.0F, 1.0F}
```

Comment:
The ranges should not be hardcoded. Instead, they should be retrieved with `fluid_settings_getnum_range()` and friends.